### ISSUE TITLE
Seems a bug for the layout for the Chinese Character

### ISSUE THREAD

**Issue by sidkang** [astanin/python-tabulate#52]
With Chinese characters in the DataFrame, it seems the tabulate package cannot identify the correct width.
```python
import pandas as pd
import tabulate

print(tabulate.__version__)
print(tabulate.WIDE_CHARS_MODE)
df = pd.DataFrame({
    '序号':['001', '021', '032'],
    '类型': ['汽车TESLA', '飞机737-MAX', 'TRAIN'],
    '备注': ['特殊', '一长串字符', '字符+1020']
})
print(tabulate.tabulate(df, tablefmt="github"))
```
The Output is :
0.8.7
True
|---|-----|-------------|------------|
| 0 | 001 | 汽车TESLA   | 特殊       |
| 1 | 021 | 飞机737-MAX | 一长串字符 |
| 2 | 032 | TRAIN       | 字符+1020  |
<img width="320" alt="Screen Shot 2020-04-21 at 12 58 57 AM" src="https://user-images.githubusercontent.com/6175895/79778431-4a544f00-836b-11ea-8f29-a8f21669d353.png">

**Comment by DrIcetar** [comment]
same issue

**Comment by jquast** [comment]
You can assign this issue to me, I believe I am working on a solution.

The problem is due to *which* version of Unicode is used by wcwidth, may be different from the version supported by your Terminal Emulator. The next version of wcwidth will support *all* versions of Unicode, and the version used by wcwidth (and tabulate) will be selected by an environment variable.

And so the solution is to only select the correct version. I am also working on a CLI utility, `wcwidth-detect`, which will automatically determine it.

**Comment by jquast** [comment]
I thought this was a unicode version issue, which I am working on a good solution for. But, there is *no issue for me to reproduce* here, at least on a Mac, using the version of `wcwidth` and `tabulate` published on pypi. Which is the same versions reported here.

I have tried `iTerm2.app`, `Terminal.app`, and XTerm version 326 on MacOS/XQuartz. I will try a few terminal emulators on Linux next.

<img width="945" alt="Screen Shot 2020-05-30 at 11 09 57 PM" src="https://user-images.githubusercontent.com/972584/83343616-b7dc8f00-a2ca-11ea-8c33-237c148c2c11.png">

**Comment by jquast** [comment]
I thought this was a unicode version issue, which I am working on a good solution for. But, there is *no issue for me to reproduce* here, at least on a Mac, using the version of `wcwidth` and `tabulate` published on pypi. Which is the same versions reported here.

I have tried `iTerm2.app`, `Terminal.app`, and XTerm version 326 on MacOS/XQuartz. I will try a few terminal emulators on Linux next.

<img width="945" alt="Screen Shot 2020-05-30 at 11 09 57 PM" src="https://user-images.githubusercontent.com/972584/83343616-b7dc8f00-a2ca-11ea-8c33-237c148c2c11.png">

**Comment by pasenor** [comment]
@sidkang can you check if you have wcwidth at all installed (`pip freeze | grep wcwidth`)? It is an optional dependency, so python-tabulate would work happily without it, but the wide character alignment would be off. For example, my output with and without wcwidth:
![image](https://user-images.githubusercontent.com/11982173/83361052-64337980-a38e-11ea-8ea6-1d99ca7373bb.png)

**Comment by sidkang** [comment]
@jquast @DrIcetar Hi, thanks for your kind reply. Sorry for not test the issue throughly. Just Checked the code in the terminal. Everything works perfectly. The Error only occurred in the Jupiter lab environment. So this could be a Jupiter side issue?
(My Jupyterlab version is 2.1.3 which is freshly installed.
<img width="825" alt="image" src="https://user-images.githubusercontent.com/6175895/83373628-a1822080-a3fb-11ea-92e3-9000f77f5ab6.png">

**Comment by jquast** [comment]
@sidkang: maybe you have multiple python environments, from your Juperlab, are you able to successfully `import wcwidth`? if not, try
```python
import subprocess, sys
subprocess.check_call([sys.executable, "-mpip", "install", "wcwidth"])
```

best wishes

**Comment by sidkang** [comment]
> @sidkang: maybe you have multiple python environments, from your Juperlab, are you able to successfully `import wcwidth`? if not, try
> 
> ```python
> import subprocess, sys
> subprocess.check_call([sys.executable, "-mpip", "install", "wcwidth"])
> ```
> 
> best wishes

Just checked the wcwidth package. Seems it is working properly.

![image](https://user-images.githubusercontent.com/6175895/83474232-1584fd00-a4be-11ea-8c55-ad84e422b733.png)

**Comment by jquast** [comment]
By the way what happened the first time the issue appeared after installing wcwidth, is you probably needed to restart your Jupyter kernel, you always need to do that when you install/upgrade packages that have already been imported

there is a kind of refresh button in the GUI for it,

but anyway it happened naturally when you stepped away for the day :) glad your issue is resolved

best wishes

**Comment by astanin** [comment]
This issue seems to be fixed now:

![image](https://user-images.githubusercontent.com/71802/108273304-464fb400-716b-11eb-9cd2-d68174649207.png)

This was fixed in pull request #58
